pkg/kubernetes/persistentvolumeclaim/v1alpha1/persistentvolumeclaim.go

```diff
@@ -73,71 +73,78 @@ func (b *listBuilder) List() *pvcList {
 
 // Len returns the number of items present
 // in the PVCList
-func (p *pvcList) Len() int {
+func (p *PVCList) Len() int {
 	return len(p.items)
 }
 
 // ToAPIList converts PVCList to API PVCList
-func (p *pvcList) ToAPIList() *v1.PersistentVolumeClaimList {
-	plist := &v1.PersistentVolumeClaimList{}
+func (p *PVCList) ToAPIList() *corev1.PersistentVolumeClaimList {
+	plist := &corev1.PersistentVolumeClaimList{}
 	for _, pvc := range p.items {
 		plist.Items = append(plist.Items, *pvc.object)
 	}
 	return plist
 }
 
 // ListBuilder returns a instance of listBuilder
-func ListBuilder() *listBuilder {
-	return &listBuilder{list: &pvcList{items: []*pvc{}}}
+func NewListBuilder() *ListBuilder {
+	return &ListBuilder{list: &PVCList{items: []*PVC{}}}
+}
+
+// NewForAPIObject returns a new instance of Builder
+func NewForAPIObject(obj *corev1.PersistentVolumeClaim) *PVC {
+	return &PVC{
+		object: obj,
+	}
 }
 
 // predicate defines an abstraction
 // to determine conditional checks
 // against the provided pvc instance
-type predicate func(*pvc) bool
+type Predicate func(*PVC) bool
 
 // IsBound returns true if the pvc is bounded
-func (p *pvc) IsBound() bool {
+func (p *PVC) IsBound() bool {
```

Comment:
can we avoid hard coded string?
```go
type PersistentVolumePhase string

const (
	// used for PersistentVolumes that are not available
	VolumePending PersistentVolumePhase = "Pending"
	// used for PersistentVolumes that are not yet bound
	// Available volumes are held by the binder and matched to PersistentVolumeClaims
	VolumeAvailable PersistentVolumePhase = "Available"
	// used for PersistentVolumes that are bound
	VolumeBound PersistentVolumePhase = "Bound"
)
```

Above is at https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/types.go.

Let us use these constants !!!

Reply:
Done